tests/materialize/materialize-snowflake/setup.sh

```diff
@@ -14,7 +14,7 @@ export SNOWFLAKE_AUTH_TYPE="${SNOWFLAKE_AUTH_TYPE}"
 # if auth type is user_password
 export SNOWFLAKE_USER="${SNOWFLAKE_USER:-}"
 export SNOWFLAKE_PASSWORD="${SNOWFLAKE_PASSWORD:-}"
-if [ -z "${${SNOWFLAKE_PRIVATE_KEY+x}}" ]; then 
```

Comment:
My shell was complaining about this syntax, so I changed it to a different syntax but with the same meaning I think

Reply:
Just a drive by comment:
This syntax really is confusing. I have no idea what the `+x` is even doing here. The [bash docs](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) cover `${var:+var}`, but the `:` seems important.

The change from `-z` to `-n` seems like it might change the meaning, though. `-z` means "is zero-length" and `-n` means "is _not_ zero-length".

Reply:
I'm not sure about the `+` syntax, but my general understanding of the intent behind this `if` block is to only load the private key if the variable is set, and that's what the new `-n` condition does: only run that block if `SNOWFLAKE_PRIVATE_KEY` is not zero-length

Reply:
I'm also remembering that I tried to get the tests to run by having `SNOWFLAKE_PRIVATE_KEY` set but set to an empty string, and then some later part of the test mechanism failing. At any rate, I'd like us to just have a way to run these tests locally using username/password auth, so whatever works for that I think is fine.